Z/IO/Process.hsc

```diff
@@ -148,11 +150,26 @@ pattern SIGHUP  = #const SIGHUP
 --   waitProcessExit pstate  -- wait for process exit on current thread.
 -- @
 initProcess :: ProcessOptions -> Resource (Maybe UVStream, Maybe UVStream, Maybe UVStream, TVar ProcessState)
-initProcess opt = initResource (spawn opt) $ \ (s0,s1,s2, pstate) -> void . forkIO $ do
-    _ <- waitProcessExit pstate
-    forM_ s0 closeUVStream
-    forM_ s1 closeUVStream
-    forM_ s2 closeUVStream
+initProcess opt = initResource (spawn opt) $
+    \ (s0,s1,s2, pstate) -> void . forkIO $ do
+        _ <- waitProcessExit pstate
+        forM_ s0 closeUVStream
+        forM_ s1 closeUVStream
+        forM_ s2 closeUVStream
+
+-- | Resourced spawn processes.
+--
+-- The same as 'initProcess', but the clean thread will try to send 'SIGTERM'
+-- to the process first.
+initProcess' :: ProcessOptions -> Resource (Maybe UVStream, Maybe UVStream, Maybe UVStream, TVar ProcessState)
+initProcess' opt = initResource (spawn opt) $
+    \ (s0, s1, s2, pstate) -> void . forkIO $ do
```

Comment:
I think we should drop forkIO here, in case of exceptions,  exceptions should be directly raised at the current thread.